### DISCUSSION TITLE
What is the meaning of the "disabling in config" term in the documentation?

### DISCUSSION THREAD

**Discussion by simon-liebehenschel** [renovatebot/renovate#18483] [Request Help] [answered]
I read in the [documentation](https://docs.renovatebot.com/self-hosted-configuration/#optimizefordisabled):

> `optimizeForDisabled`
>
> Set to true to first check for disabling in config before cloning.

I can read and understand English, but I completely do not understand this short and meaningless description of the `optimizeForDisabled`.

Check for what? For "disabling"? What is "disabling" in this context? What can be disabled? How is this relevant to cloning?

I would try to debug the code to see what this parameter do, but sorry, I develop only in Python.

**Comment by RahulGautamSingh** [comment, marked as answer]
Hi @AIGeneratedUsername ,

Renovate typically clones your repository and examines the config-file to determine whether or not we need to process it by looking for the `enabled` option.
However, if you set `optimizeForDisabled` to true, Renovate will use the API to retrieve the renovate-config file check `enabled` option in it, before cloning, which is good optimization for big repos.
Here is a [link ](https://github.com/renovatebot/renovate/issues/3934) to know more about why the option was added.

Thanks for letting us know that it was hard to understand, we will update the docs soon.

**Reply by viceice** [reply]
By default it only checks `renovate.json` in repo root.

https://github.com/renovatebot/renovate/blob/867c47ff6542fa36789b574f216f31aa13cdc0a8/lib/workers/repository/init/apis.ts#L28-L37

Self-hosted users can change that name via `onboardingConfigFileName`
https://github.com/renovatebot/renovate/blob/867c47ff6542fa36789b574f216f31aa13cdc0a8/lib/workers/repository/init/apis.ts#L15-L18

https://github.com/renovatebot/renovate/blob/867c47ff6542fa36789b574f216f31aa13cdc0a8/lib/config/app-strings.ts#L1-L11
